From c89600a635eb6871059de6b2eadedeebf1b5909c Mon Sep 17 00:00:00 2001
From: Ondrej Jirman <megi@xff.cz>
Date: Sun, 21 May 2023 01:26:59 +0200
Subject: [PATCH 397/406] input: touch: goodix: Try to keep regulator
 enable/disable balanced

Failure handling is still not optimal in this driver, but let's
try to improve it.

Signed-off-by: Ondrej Jirman <megi@xff.cz>
---
 drivers/input/touchscreen/goodix.c | 1 +
 1 file changed, 1 insertion(+)

diff --git a/drivers/input/touchscreen/goodix.c b/drivers/input/touchscreen/goodix.c
index e9c3ad164950..fe06a19c0af3 100644
--- a/drivers/input/touchscreen/goodix.c
+++ b/drivers/input/touchscreen/goodix.c
@@ -1505,6 +1505,7 @@ static int goodix_resume(struct device *dev)
 
 		error = regulator_enable(ts->vddio);
 		if (error) {
+			regulator_disable(ts->avdd28);
 			dev_err(dev, "Regulator vddio enable failed.\n");
 			return error;
 		}
-- 
2.34.1

